matching-notification-api/src/main/java/org/phenotips/matchingnotification/internal/DefaultMatchingNotificationManager.java

```diff
@@ -86,14 +87,17 @@
 
         for (PatientMatchEmail email : emails) {
 
-            List<PatientMatchNotificationResponse> notificationResults = this.notifier.notify(email);
+            PatientMatchNotificationResponse notificationResult = this.notifier.notify(email);
 
-            List<PatientMatch> successfulMatches = this.getSuccessfulNotifications(notificationResults);
-
-            this.updateNotificationHistory(successfulMatches, email, "notification");
+            if (notificationResult.isSuccessul()) {
+                Collection<PatientMatch> updatedMatches =
+                    this.updateNotificationHistory(notificationResult, email, "notification");
+                notificationResult.setPatientMatches(updatedMatches);
```

Comment:
this whole design where we pass `notificationResult` into a method, get something back, which is then only used to update `notificationResult` outside the method looks iffy, updating it inside `updateNotificationHistory()` seems more proper. Not critical though